src/sync/utils.rs

```diff
@@ -97,7 +97,10 @@ macro_rules! client_request {
 /// The context of ttrpc (sync).
 #[derive(Debug)]
 pub struct TtrpcContext {
+    #[cfg(unix)]
     pub fd: std::os::unix::io::RawFd,
+    #[cfg(windows)]
+    pub fd: i32,
```

Comment:
I think it's better to wrap the fd struct, such as:
```
struct RawFd {
#[cfg(unix)]
    pub fd: std::os::unix::io::RawFd,
    #[cfg(windows)]
    pub fd: i32,
}
```

Reply:
would this change the public api?

Reply:
This does change the public API, @Tim-Zhang  have any suggestions?

Reply:
That's why I did it this way, otherwise it would be pretty big cascading changes I think

Reply:
I understand what you are concerned about, but I think that window support is a major feature, whether it can be upgraded to a major version, maybe the change of API is also a matter of course.

Reply:
If I remember correctly, this type of change would require updates to the codegen (and maybe the compiler?).  

My inclination is to hold off on changes to the public API since the project is still < 1.0 and I would like to add windows async support as well.  

In addition, we might learn more about the shape of the API especially as the Windows side gets used more?

Reply:
Ok, I agree with you.